src/scheduler.py

```diff
@@ -155,6 +202,163 @@ def log_route(
         if log_level == "debug":
             logger.debug(fmt_string, str(route), *args)
 
+    async def query_denom_route(
+        self, query: DenomRouteQuery
+    ) -> Optional[list[DenomRouteLeg]]:
+        if self.denom_routes and query in self.denom_routes:
```

Comment:
do we exit early here in case we already have a response stored? would be nice to add a comment